consensus/src/chained_bft/consensus_types/block.rs

```diff
@@ -359,8 +431,13 @@ where
         proto.set_round(self.round());
         proto.set_height(self.height());
         proto.set_quorum_cert(self.quorum_cert().clone().into_proto());
-        proto.set_signature(self.signature().to_bytes().as_ref().into());
-        proto.set_author(self.author.into());
+        if let BlockSource::RealBlock { author, signature } = self.block_source {
+            proto.set_signature(signature.to_bytes().as_ref().into());
+            proto.set_author(author.into());
+            proto.set_is_nil_block(false);
+        } else {
+            proto.set_is_nil_block(true);
```

Comment:
probably don't need this field? we can tell from whether the author/sig field is empty

Reply:
Well, that's interesting: today the proto definition of `author` and `signature` is just `bytes`, hence determining "emptiness" is problematic: unlike other fields that can be `Option`, the `author` and `signature` fields are converted to `::bytes::Bytes` in Rust. Hence I had to add a separate boolean field to distinguish between the two.

Reply:
Bytes has a function is_empty() so I thought it's enough

Reply:
Yea, I somehow suspected the `is_empty()` to be fragile. In the current implementation the `default()` of bytes generates an empty array, so I guess we could use that...